packages/psss/src/containers/Panels/AlertsPanel.js

```diff
@@ -67,74 +67,81 @@ const menuOptions = [
-    const [isModalOpen, setIsModalOpen] = useState(false);
-    const [activeIndex, setActiveIndex] = useState(0);
-    const sitesState = useFetch(getAffectedSites);
-    const notesState = useFetch(getAlertsMessages);
-    const activityState = useFetch(getActivityFeed);
-    const countryName = useSelector(state => getCountryName(state, countryCode));
+const TabsContext = createContext(null);
 
-    const handleChange = option => {
-      // todo handle changes other than creating an outbreak
-      setIsModalOpen(true);
-    };
+export const AlertsPanelContext = createContext(null);
 
-    return (
-      <Drawer open={isOpen} onClose={handleClose}>
-        <DrawerTray heading="Alert Details" onClose={handleClose} Icon={WarningCloud} />
-        {period && countryCode && (
-          <AlertsDrawerHeader
-            dateText={`Triggered on: W${getWeekNumberByPeriod(period)}`}
-            date={getDisplayDatesByPeriod(period)}
-            avatarUrl={countryFlagImage(countryCode)}
-            subheading={countryName}
-            heading={syndromeName}
-            DropdownMenu={<DropdownMenu options={menuOptions} onChange={handleChange} />}
-          />
-        )}
-        <TabsContext.Provider value={{ activeIndex, setActiveIndex }}>
-          <CardTabList Context={TabsContext}>
-            <CardTab>
-              <LocationOn /> Affected Sites
-            </CardTab>
-            <CardTab>
-              <SpeakerNotes />
-              Notes ({notesState.count})
-            </CardTab>
-            <CardTab>
-              <List />
-              Activity
-            </CardTab>
-          </CardTabList>
-          <CardTabPanels Context={TabsContext}>
-            <AffectedSitesTab state={sitesState} />
-            <NotesTab state={notesState} />
-            <ActivityTab
-              state={activityState}
-              NotesTabLink={<LinkButton onClick={() => setActiveIndex(1)}>note</LinkButton>}
-            />
-          </CardTabPanels>
-        </TabsContext.Provider>
-        <CreateOutbreakModal isOpen={isModalOpen} handleClose={() => setIsModalOpen(false)} />
-      </Drawer>
-    );
-  },
-);
+export const AlertsPanelProvider = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [data, setData] = useState({});
 
-AlertsPanel.propTypes = {
-  period: PropTypes.string,
-  countryCode: PropTypes.string,
-  syndromeName: PropTypes.string,
-  isOpen: PropTypes.bool.isRequired,
-  handleClose: PropTypes.func.isRequired,
+  return (
+    <AlertsPanelContext.Provider value={{ data, setData, isOpen, setIsOpen }}>
+      {children}
+    </AlertsPanelContext.Provider>
+  );
 };
 
-AlertsPanel.defaultProps = {
-  period: '',
-  countryCode: '',
-  syndromeName: '',
+AlertsPanelProvider.propTypes = {
+  children: PropTypes.node.isRequired,
 };
+
+export const AlertsPanel = React.memo(() => {
+  const { data: panelData, isOpen, setIsOpen } = useContext(AlertsPanelContext);
```

Comment:
The context is a nice solution to making the button which opens the panel and the panel share the state of the alert. Hopefully the actual data for the panel can be just fetched using react-query and so it doesn't have to be passed through context. It's a bit hard to see until it's finished if panel data is for the reference to the alert data or the actual alert data.